packages/clarity-js/src/layout/dom.ts

```diff
@@ -42,18 +43,27 @@ function reset(): void {
     idMap = new WeakMap();
     regionMap = new WeakMap();
     iframeMap = new WeakMap();
-    if (Constant.DEVTOOLS_HOOK in window) { window[Constant.DEVTOOLS_HOOK] = { get, getNode, history }; }
+    maskedMap = new WeakMap();
+    if (Constant.DevHook in window) { window[Constant.DevHook] = { get, getNode, history }; }
 }
 
-export function extractRegions(root: ParentNode): void {
-    for (let key in config.regions) {
-        // We check for regions in the beginning (document) and later whenever there are new additions or modifications to DOM (mutations)
-        // Since mutations may happen on leaf nodes too, e.g. textnodes, which may not support all selector APIs.
-        // We ensure that the root note supports querySelector API before executing the code below to identify new regions.
-        if (config.regions[key] && "querySelector" in root) {
+// We parse new root nodes for any regions or masked nodes in the beginning (document) and 
+// later whenever there are new additions or modifications to DOM (mutations)
+export function parse(root: ParentNode): void {
+    // Since mutations may happen on leaf nodes too, e.g. text nodes, which may not support all selector APIs.
+    // We ensure that the root note supports querySelector API before executing the code below to identify new regions.
+    if ("querySelector" in root) {
+        // Extract regions
+        for (const key of Object.keys(config.regions)) {
             let element = root.querySelector(config.regions[key]);
             if (element) { regionMap.set(element, key); }
         }
+
+        // Extract nodes with explicit masked configuration
+        for (const entry of config.suppress) {
+            let element = root.querySelector(entry);
+            if (element) { maskedMap.set(element, true); }
+        }
```

Comment:
what happens when mask is true and suppress is also set in config? : do we need to think about unmasking some elements only based on config?